File: rb/features.rb
# HtmlToPdfConverter SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module HtmlToPdfConverterFeatures
  def self.make_feature(name)
    case name
    when "base"
      HtmlToPdfConverterBaseFeature.new
    when "ratelimit"
      HtmlToPdfConverterRatelimitFeature.new
    when "retry"
      HtmlToPdfConverterRetryFeature.new
    when "test"
      HtmlToPdfConverterTestFeature.new
    when "timeout"
      HtmlToPdfConverterTimeoutFeature.new
    else
      HtmlToPdfConverterBaseFeature.new
    end
  end
end

File: py/features.py
# HtmlToPdfConverter SDK feature factory

from feature.base_feature import HtmlToPdfConverterBaseFeature
from feature.test_feature import HtmlToPdfConverterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HtmlToPdfConverterBaseFeature(),
        "test": lambda: HtmlToPdfConverterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
